test/sfs.py

```diff
@@ -43,11 +102,20 @@ def seek(self, pos, whence=0):
             raise ValueError("illegal value of whence")
 
     def read(self, size):
-        res = c_helper.read_file_py(self._fp, self._pos, size)
+        buffer = ctypes.create_string_buffer(size)
+        read_status = c_helper.nfs4_pread(self._file, buffer, self._pos, size)
+        if read_status != NFS4_OK:
+            print("Failed to read file: " + c_helper.nfs4_error_string(client))
+            return
         self._pos += size
-        return res
+        return bytearray(buffer)
```

Comment:
Does this create a copy? Would like to avoid copies if possible.

Reply:
addressed!